### DISCUSSION TITLE
Change in icons made it harder to use the toolbar(?)

### DISCUSSION THREAD

**Discussion by rexelbartolome** [freeplane/freeplane#326] [Ideas]
So with the change in icons, I'm aware that modernizing and making them consistent came at a cost of variety and ease in finding the specific button you want to use in the toolbar. To anyone affected significantly by this change, you can use this thread to give your thoughts :)

With this, I have a few solutions in mind:

### Recolor: 
a dedicated LaF/theme that uses a specific set of recolored icons for the toolbar. A fair bit of work but I feel like it's a band aid solution to a UI problem. 

### Reorganization: 
working on how the toolbar is organized so that it's easier to use, despite the color consistency. Lots of work in both visuals and coding I think.

Not sure which solution is tougher to execute (or if even both of them are possible to do) but I'm leaning towards reorganization.

I think one part that contributes to it is we just have too many icons/buttons and some of those high-traffic buttons are right beside the really underused ones. Also Filter buttons are organized horizontally, meaning the filter button you want to use can be on either side of the screen. 

So this is what I thought of if I were to declutter and reorganize the toolbar (i haven't added the filter buttons just yet)

![image](https://user-images.githubusercontent.com/88552647/164883765-aaeb8ca6-2190-4a4a-9d55-87a32cfb6045.png)

### Explanation

**Viewing/editing button** is now hidden under a submenu so new users won't be able to lock themselves out of their own map

**New File button** is too near to `File > New File` therefore it's removed. This is usually how other programs work anyway: New File isn't visible on the toolbar, you have to click it under the File menu

**Next and Prev mindmap** is a bit redundant, you already see the row of mindmaps you have open anyway. Much bigger space to click, more accurate to click the map that you actually want to go to instead of flipping through them

**Close current map button** is redundant as well, we currently have two other close buttons visible besides this one, so we can lose one out of three of them.

**Select chain** I think is underused and has a very situational use, feel free to correct me on that. 

**Print button** is also underused and situational, it's alright that we put that inside a submenu under File

**Find, Find and Replace and Filtering** I think need a bit of a rework in terms of what buttons are visible on their respective popup windows. Either way, they can be organized on the right side so they can be organized vertically and horizontally so all the filter stuff are reliably on the right side of your screen.

**Zoom level** would be moved to bottom right to be in line with other programs

Don’t have a solid idea regarding the **Add Node button** but maybe it could have a dropdown menu of the different types of nodes you can insert/create. 

**Formatting buttons** and their drop down menus can still be reorganized once the filter stuff is taken into account. Just want to know some feedback before getting into them

As always, let me know your thoughts :D

**Comment by dpolivaev** [comment]
I like the discussion, removing some less used elements from the toolbar might help. I could add code to collect statistics about how often the icons are used, but I can also trust your judgment or make it configurable. (However, I remember quickfold wrote about Microsoft research showing that only a minority of users modify default UI elements).

Because some users have wide monitors, it should be possible to show all toolbars also in one line if there is enough horizontal space. I think it should be considered in your mocks as well. Therefore I am not sure if the use of double-height icons is necessary.

**Reply by rexelbartolome** [reply]
Yea I didn't plan on making it configurable. Adding some statistics could be a good way to have a more informed opinion regarding what buttons are most used. 

Hmmmm I was thinking maybe the Filter Toolbar can already be visible by default since we have more room to work with anyway, so the "Basic" Toolbar and the Filter Toolbar would be both visible and I think both of them would be enough occupy the whole width of a normal screen. Not sure about Ultra Wide ones though

**Reply by EdoFro** [reply]
I think the Filter Toolbar should not be visible by default. It's very powerful but pretty intimidating for new users, don't you think?
I don't think it's a good idea to new users to see it at the first glance.

**Comment by EdoFro** [comment]
hi @rexelbartolome,
Just as information for your work, I marked in the following image which are the buttons that I really use from the toolbar:

![image](https://user-images.githubusercontent.com/35700575/165190616-1d1106a6-2e7b-4d16-a62b-6dd951d91376.png)
 
and just a comment:

> **Select chain** I think is underused and has a very situational use, feel free to correct me on that.

I use it not so often, but when I need it I'm very pleased that it's right there in the toolbar

I like your proposal. 

bye,
edo

**Reply by rexelbartolome** [reply]
ohhh that's actually really useful to know, thanks :) i'll explore more on the filter buttons if i can put them in a more toggleable layout

**Reply by adelintru** [reply]
I would love that. You can customize the buttons in the main toolbar, but not in the other. A toggleable layout would simply be a dream come true.

**Comment by bunk3m** [comment]
I updated today and am disappointed with the new icons. It's quite hard to read for my old eyes.

As @nbruno78 noted in another thread, I also liked the older icons better.  I'll try to use the zip file from @abc16361 for the older icons.  

While I understand that adding an icon pack to the UI creates extra work over the long term, I find the new icons to be too hard to distinguish and are not obvious for what they do.  :-(

**Reply by kamilvlcek** [reply]
I feel quite confused by the new set of icons. I have to search repeatedly what each icon means and under which icon I find the action I want. The most difficult is they are all the same colour, so they are very similar to each other. Then, mostly their position on the toolbar can be used to find them quickly. Basically, I'll probably stop using most of them. 
What about enabling multiple sets of toolbar icons and switching between them? One set can be called a theme, or so.

**Reply by dpolivaev** [reply]
The majority of people prefer old, well-known, and trusted environments and designs. Accepting the change costs effort and takes time. Does it mean that old designs should be frozen and never changed? I don't think so. I was thinking a lot about it. Currently, I believe that the old icons were easier to find just because of the habit. The new icon set is expressive and consistent, and I got used to it after a very short time. Maintaining both icon sets would cost work, it would let Freeplane lose its new graphical language and identity and make new documentation harder to create and use. So even feeling and understanding the pain of the old users I believe that going back and creating themes are both no good options.

**Comment by rexelbartolome** [comment]
Hello, here's an update with the reorganization. I realized that one of the big problems with how FP is currently designed is the top section/toolbar is overburdened with a lot of features with search and filter being one of the primary uses of it. This becomes a high traffic area for super users and kind of a barrier for new users since a lot of powerful features are presented to them without explanation. 

For comparison Xmind only has 11 buttons on it's top area 
![image](https://user-images.githubusercontent.com/88552647/168979017-0a0085b8-944a-4d56-9693-01b22652c805.png)

Mindmeister has just 11 buttons too
![image](https://user-images.githubusercontent.com/88552647/168979176-4c9cbcdd-d14c-4106-bc82-89df73389844.png)

Freeplane currently has 50+ buttons (with the filter and search bar enabled)

A more elegant solution would be to put other features in other areas of the screen, a great example is the Formatting panel. It's located on the right side and pretty much anything you can think of that you want to edit, formatting wise, is on that panel. Removing the burden from the top toolbar and having the freedom for the toolbar formatting options to be more simple and concise.

![image](https://user-images.githubusercontent.com/88552647/168980895-3c9e7c44-89aa-49dd-894a-7e3e4b950d4b.png)


Another way (that has also been recently discussed) would be to put the search/filter section and have it as panel (perhaps on the left side of the screen, letting the left panel have a tab for search, filter, and icons) which is consistent with how other applications treat the search/replace feature too.

This is Visual Studio Code's search and replace menu
![image](https://user-images.githubusercontent.com/88552647/168980273-7269e231-fdbf-4d7a-9326-644ff17f94b8.png)

This is Xmind's
![image](https://user-images.githubusercontent.com/88552647/168980416-56afd918-c400-48ce-b7ef-faee724461ff.png)

But I digress, of course, making such a sweeping change in the UI/UX of such a legacy program like Freeplane can be jarring both for the user and the developer, I just wanted to share my thoughts on it.

Here's what the reorganization looks like right now
![image](https://user-images.githubusercontent.com/88552647/168977763-2853a3be-90b2-4748-b139-e61ce8bd97c7.png)

refined the sections a bit more. There's still a lot of empty space that isn't being utilized (especially in the search section) which is why even though I removed a lot of icons, the end result is only 5% shorter. I'll look into experimenting how the layout can change so we can utilize that empty space, differentiating the sections even more.

![Photoshop_ORd96GNHtC](https://user-images.githubusercontent.com/88552647/168977796-ff57b147-b78a-48e9-b3a5-a0fc3e78cb3b.png)
![YLOoujVNmJ](https://user-images.githubusercontent.com/88552647/168977811-4ed5d3ce-d749-43e0-9bc3-c9b8c2a5904a.png)
![Photoshop_85yelqEDJd](https://user-images.githubusercontent.com/88552647/168977805-13d54d9a-e237-4916-a4d0-5918dcd1cd50.png)

as always let me know your thoughts :)

**Reply by euu2021** [reply]
I agree with your conclusion about the top section/toolbar being overburdened. About that, I will bring back my [Basic Bar hypothesis](https://sourceforge.net/p/freeplane/discussion/758437/thread/60ba129980/): keep a complete bar as an option, but, by default, have a basic bar designed having in mind new users and minimalist users. 

On a side note, your first screenshot shows a Xmind map where one of the nodes has a good-looking node with a big sticker and a text below it. Let's remember that this is something that FreePlane still can't do, and I think would bring a good improvement to possibilities of map design, as I explained [here](https://sourceforge.net/p/freeplane/discussion/758437/thread/66dd19d25a/).

Anyway, I like your proposal of two lines of icons.

**Reply by macmarrum** [reply]
@rexelbartolome 👍 I very much like the idea to put the controls on two lines, bringing buttons closer to each other. When going for a button, I can use a 2-step search approach, treating each section as a first-level look-up. E.g. to locate `bold`, I know I'd have to look in the section left of center, where font formatting controls are placed. Once I arrive there, I can consider changing font family or size too. In other words, it makes the process of discovering possibilities much easier. Especially for people new to the tool and its user interface.

**Reply by adelintru** [reply]
I very much like this discussion as the toolbar is indeed too busy. My perspective is that allowing the user to change the filter toolbar like you can change the main one (via the ini file) is the simplest solution. I am not suer how difficult it would be to have a windows which makes these ini changes more usefriendly.

Maybe a quick question: is there a ressource showing all commands/buttons that can be added to the main toolbar? Maybe the quickest workarround is to add the search buttons to the main toolbar. Thank you

**Reply by dpolivaev** [reply]
@rexelbartolome could you please send me the "invert toggle button" as SVG?
I have implemented your suggestions and only need the button.

**Reply by rexelbartolome** [reply]
@dpolivaev Awesome :) The invert toggle icon is named as `NotFilter.svg` and is already used alongside the other Filter icons. 
![image](https://user-images.githubusercontent.com/88552647/172028667-49af1cf0-2762-4c4e-ad3a-3591bd9b4b38.png)


Regardless, it's here:

[NotFilter.zip](https://github.com/freeplane/freeplane/files/8839086/NotFilter.zip)

**Comment by dpolivaev** [comment]
Try preview 1.10.3_08 and give feedback

**Reply by dpolivaev** [reply]
Buttons with submenus are still on my todo list.

**Reply by rexelbartolome** [reply]
Awesome!
Here's my feedback regarding the formatting area

## 1.10.3 Preview vs Proposal
https://user-images.githubusercontent.com/88552647/172089889-2762aae7-35d8-4c88-8b5d-999a2f0907d5.mp4


## Proposal
![Photoshop_wSX3VxUPA2](https://user-images.githubusercontent.com/88552647/172089929-8819fd36-72a8-404a-8a2f-ea64be68c2fa.png)


So from my understanding, we have two invisible containers and the dropdown menus extend to the edges of those invisible containers. So I propose that the containers have a grid system of sort, based on how many icons it could fit inside.

So the red boxes are `icon boxes` that we'll use for measurement. I'll just refer to them as boxes. 

The `Style` dropdown is currently around 8 boxes long, we could shorten this to 4, having just one empty box as allowance for longer style names

https://user-images.githubusercontent.com/88552647/172089855-8c5cfb34-787b-4099-a801-6c8691eed6a6.mp4

The `font face / font style` dropdown is currently around 8.5 boxes long which could be shortened to 7 boxes. Then the `font size` dropdown is currently 2.5 boxes long, which we could shorten to 2 boxes instead, to be more inline with the grid. Then we'll just have one empty box between the font size and the other icons

https://user-images.githubusercontent.com/88552647/172089871-e38af5d2-a40b-4d3c-928f-db7a43d2a8e2.mp4

---

These changes don't necessarily have to be aligned pixel-perfect to 'x' amount of boxes etc. Just figured it would be a better way of thinking about the space

For the other sections, I still need to think about them 🤔 and I'll just post here again when the time comes

**Reply by dpolivaev** [reply]
I improved it, check 1.10.3_09

**Reply by rexelbartolome** [reply]
The improvement looks great, thanks :)



# Search Section (the proposal is on top, the current preview is below)

I think the find next and find previous should be next to the search bar, as it's standard for other applications

https://user-images.githubusercontent.com/88552647/172976717-25a9f8fa-607d-44a1-800e-80fbc44f08ef.mp4

I think a lot of the feedback mentioned was the change in icons made them look too similar with one another with no clear indication of what they do. And I kind of agree, we have a lot of `small magnifying glass icon with blue symbol on the top right` and `small filter icon with blue symbol on top`

So to combat that, I think we can use more generic but clearer icons. So for this one, `find previous` and `find next` should use `generic_forward.svg` and `generic_back.svg`. Those SVG's are already inside the current build as far as I know, but here they are either way
## [forwardback.zip](https://github.com/freeplane/freeplane/files/8875273/forwardback.zip)

![Photoshop_t4q1cLj7Fz](https://user-images.githubusercontent.com/88552647/172976823-c6fd66c6-9402-4213-9e6d-aa30c86bb8cb.png)

And I think we should also remove the `find next` and `find prev` in the Home Section but have those buttons available inside the popup window

![Photoshop_O7KsLs7K0C](https://user-images.githubusercontent.com/88552647/172976836-54d514ac-9e57-43c8-a7ec-f337599706a9.png)

# Filter Section

For this one, bit of the same thing, rearranged the undo and redo filter action buttons and made them use the same svg as the normal one's named `undo.svg` and `redo.svg`. I don't think there's going to be a confusion between them since normal undo is on the leftmost part of the toolbar and the undo filter action is on the rightmost part of the toolbar

https://user-images.githubusercontent.com/88552647/172976724-e5a6a1dc-beaa-4c49-b728-394ae6f7415c.mp4



![Photoshop_VIrIQlSmhN](https://user-images.githubusercontent.com/88552647/172976808-eab0fbb5-ef2f-445b-9ecd-c3ab8bb856c2.png)

# Toggleable Buttons

And finally the toggleable buttons lost their background, I was hoping to keep those to differentiate which ones are single-use buttons and which ones are toggleable buttons that can be activated.


![Photoshop_hp7mBY5FWM](https://user-images.githubusercontent.com/88552647/172976855-f1826ec0-825c-40fd-a3d0-a7cda48a6ccd.png)


they're supposed to look like this 

![image](https://user-images.githubusercontent.com/88552647/172979430-feb14e13-8ce4-4286-a4ce-548d8550e9c6.png)

**Reply by dpolivaev** [reply]
> And finally the toggleable buttons lost their background, I was hoping to keep those to differentiate which ones are single-use buttons and which ones are toggleable buttons that can be activated.

I don't understand it. How was it before? When was is different? What exactly is lost?

**Reply by dpolivaev** [reply]
All other changes are available in preview 1.10.3_10

**Comment by EdoFro** [comment]
Dear @rexelbartolome,
I like the new icons, but today I noticed that when working in a notebook with a 14 inch screen, some of them look very similar at first sight.

For example:
![image](https://user-images.githubusercontent.com/35700575/176940681-a3d1c13c-6602-4d8c-89ab-427c1ec29a06.png)

The concept in each one is correct: I easy understand what they mean.
But when not working in a place with good illumination or being old and you can't adapt so easy the focus in your eyes, they are too similar from each other.
I have to look at them for a moment or wait to the tooltip to appear to be sure I click the right one.

I don't know how they could be better, but I wanted to give you this feedback.

best wishes,
edo

**Reply by Illegit** [reply]
Rexel maybe inverting the sizes big blue cross with small dark funnel, ... can help

**Reply by dpolivaev** [reply]
You could try to increase the icon size, it is configurable at Preferences…->Appearance->Look and feel->Toolbar icon height

**Reply by dpolivaev** [reply]
But I agree that it would be nice if @rexelbartolome could find time to create icons with larger key symbols

**Reply by Illegit** [reply]
Hi Rexel, if you have the time and possibility, may I suggest that you review all of these icons in terms of 2 parameters:
function & context and attributing a color/size always the same for all of the functions and so on for the contexts... this probably could help the users to associate an icon to command without starting a trial & error interpretation.
About the global icon size, I follow the Dimitry indication.
Finally I think these icons we have now in FP are a drastical improvement and now it is a matter of fine-tuning to help the final users.
Thankyou for your work. that I use every single day.
illegit

**Reply by rexelbartolome** [reply]
Thanks for the feedback :D Yes the problem is they look too similar to each other, both in color and silhouette which is a problem for legibility, not just for people with bad eyesight, but for smaller displays as well etc. 

I'm not sure if making a new icon is necessary, I think reusing some of the other (more distinct) icons can be used so they have a much different shape. It's very similar to the other icon problems with `find next`, `find previous`, `undo` and `redo filter action`, so maybe a similar solution could work too. I'll try looking for a more appropriate one in the existing icon set :)

**Reply by rexelbartolome** [reply]
Can't seem to find a suitable icon, so I just used existing parts of other icons to make new one for the `compose filter` button. Along with that, I also replaced the `reapply filter action`. I think the `no filtering` button can be replaced with `cancel_button.svg` which freeplane already has in its svg folder.
![image](https://user-images.githubusercontent.com/88552647/178104487-f62ccccc-3648-419b-9fcc-876ab058dcb9.png)

[filterbuttons.zip](https://github.com/freeplane/freeplane/files/9076417/filterbuttons.zip)

so basically,
replace `edit_filtering_condition.svg` and `reapply_filter.svg` with the ones in the filterbuttons zip file. 
Then inside the freeplane.prop: instead of `remove_filtering.svg` for `ApplyNoFilteringAction.icon`, use `cancel_button.svg`

**Reply by dpolivaev** [reply]
![Selection_117](https://user-images.githubusercontent.com/1558052/178106059-6367be62-1393-4287-b65d-611c914b23b3.png)

Thanks, icons replaced
